core/crypto/secp256k1/impl/secp256k1_sha256_provider_impl.hpp

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CPP_FILECOIN_CORE_CRYPTO_SECP256K1_IMPL_SECP_256_K_1_SHA_256_PROVIDER_IMPL_HPP
+#define CPP_FILECOIN_CORE_CRYPTO_SECP256K1_IMPL_SECP_256_K_1_SHA_256_PROVIDER_IMPL_HPP
+
+#include "crypto/secp256k1/impl/secp256k1_provider_impl.hpp"
+
+namespace fc::crypto::secp256k1 {
+
+  /**
+   * Implementation of secp256k1 provider with sha256 digest function
+   */
+  class Secp256k1Sha256ProviderImpl : public Secp256k1ProviderImpl {
```

Comment:
Am I right, that this provider can use any digest algorithm returning Hash256?
If yes, Blake fits this requirement.
So maybe make this provider a template class taking digest algorithm as parameter?

Reply:
Yes, but we don't need bls hash.